### PR TITLE
Do not ignore trailing character after leaf. Fix ROOT-8577

### PR DESCRIPTION
**Pull request by pcanal**

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
@bellenot Do you understand why:

    projectroot.runtutorials.tutorial_graphics_tmathtext
    projectroot.runtutorials.tutorial_legacy_rootenv

fail on windows as a result of this PR?

**Comment by bellenot** [comment]
> @bellenot Do you understand why:
> 
> ```
> projectroot.runtutorials.tutorial_graphics_tmathtext
> projectroot.runtutorials.tutorial_legacy_rootenv
> ```
> 
> fail on windows as a result of this PR?

Well, it's probably unrelated, I see also a couple of failures in the master (nightly):
```
02:47:27 The following tests FAILED:
02:47:27 	403 - tutorial-graphics-tmathtext (Failed)
02:47:27 	516 - tutorial-legacy-rootenv (Failed)
```

File: tree/treeplayer/src/TTreeFormula.cxx [hunk 1]
```diff
@@ -1184,6 +1184,12 @@ Int_t TTreeFormula::ParseWithLeaf(TLeaf* leaf, const char* subExpression, Bool_t
             }
          }
       }
+   } else {
+      // Regular/old TLeaf, there should be anything afterward ...
```
Comment (hageboeck) [inline tree/treeplayer/src/TTreeFormula.cxx:1188]:
```suggestion
      // Regular/old TLeaf, there should not be anything afterward ...
```

**Review by hageboeck — APPROVED** [review]
Maybe a test would be nice, but the code certainly looks like it's doing what the commit describes.

**Comment by pcanal** [comment]
The test can be seen in https://github.com/root-project/roottest/pull/584

**Comment by pcanal** [comment]
The macos failures are unrelated:
```
<<< cling interactive line includer >>>: fatal error: module file '/build/jenkins/workspace/root-pullrequests-build/build/lib/MathCore.pcm' is out of date and needs to be rebuilt: module file out of date
<<< cling interactive line includer >>>: note: imported by module 'MathMore' in '/build/jenkins/workspace/root-pullrequests-build/build/lib/MathMore.pcm'
Module 
```

**Pull request merged**